specs/spec.go

```diff
@@ -11,8 +11,10 @@ import (
 type Kind int
 
 type Spec struct {
-	Kind Kind        `json:"kind"`
-	Spec interface{} `json:"spec"`
+	Kind   Kind        `json:"kind"`
+	Plugin interface{} `json:"plugin"`
+
+	Warnings []string `json:"-"`
```

Comment:
Another option, if we don't like `Warnings` on this struct (which is a bit weird, I agree), but want to maintain backwards-compatibility in the SDK, is we can return a custom Error object with a `Warnings` attribute from `SpecUnmarshalYamlStrict`

Reply:
`error` but not an error? that's also equally weird (and even unclear maybe?)